postgres_composite_types/__init__.py

```diff
@@ -249,23 +258,8 @@ def register_composite(cls, connection, retry=False):
         to register itself again after the type is created.
         """
 
-        try:
-            with connection.temporary_connection() as cur:
-                register_composite(cls._meta.db_type, cur, globally=True)
-        except ProgrammingError:
-            if retry:
-                composite_type_created.connect(post_migrate_register,
-                                               sender=cls)
-            else:
-                raise
-
-
-def post_migrate_register(signal, sender, connection, **kwargs):
-    """
-    Register a CompositeType with Postgres after it has been migrated.
-    """
-    sender.register_composite(connection, retry=False)
-    composite_type_created.disconnect(post_migrate_register, sender=sender)
+        with connection.temporary_connection() as cur:
```

Comment:
From [the docstring](https://github.com/django/django/blob/a5c8072ab1e56da368a40f3e9c9a3b465f4ffbae/django/db/backends/base/base.py#L554) (it does not appear to be documented in the Django docs):
```
Context manager that ensures that a connection is established, and
if it opened one, closes it to avoid leaving a dangling connection.
This is useful for operations outside of the request-response cycle.

Provides a cursor: with self.temporary_connection() as cursor: ...
```

Saves me having to manually create and close a connection for a single query like this. It is used through out the database drivers for simple things like querying the database version.